pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/when/impl/BuildingChangeConditional.java

```diff
@@ -0,0 +1,347 @@
+import org.apache.commons.lang.StringUtils;
+import org.apache.tools.ant.types.selectors.SelectorUtils;
+import org.codehaus.groovy.ast.expr.Expression;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.pipeline.modeldefinition.ast.ModelASTWhenContent;
+import org.jenkinsci.plugins.pipeline.modeldefinition.parser.ASTParserUtils;
+import org.jenkinsci.plugins.pipeline.modeldefinition.when.DeclarativeStageConditional;
+import org.jenkinsci.plugins.pipeline.modeldefinition.when.DeclarativeStageConditionalDescriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import javax.annotation.CheckForNull;
+import java.io.File;
+
+import static org.apache.commons.lang.StringUtils.defaultIfBlank;
+import static org.apache.commons.lang.StringUtils.isNotEmpty;
+import static org.jenkinsci.plugins.pipeline.modeldefinition.when.impl.BuildingChangeConditional.Names.*;
+
+public class BuildingChangeConditional extends DeclarativeStageConditional<BuildingChangeConditional> {
+
+    private String id;
+    private String target;
+    private String branch;
+    private String fork;
+    private String urlX;
+    private String title;
+    private String titleX;
+    private String author;
+    private String authorDisplayName;
+    private String authorX;
+    private String authorDisplayNameX;
+    private String authorEmail;
+    private String authorEmailX;
+
+    @DataBoundConstructor
+    public BuildingChangeConditional() {
+    }
+
+    /**
+     * CHANGE_ID ant style matching.
+     * @return id
+     * @see ChangeRequestSCMHead#getId()
+     */
+    public String getId() {
+        return id;
+    }
+
+    @DataBoundSetter
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    /**
+     * CHANGE_TARGET ant style matching.
+     * @return target
+     * @see SCMHead#getName()
+     */
+    public String getTarget() {
+        return target;
+    }
+
+    @DataBoundSetter
+    public void setTarget(String target) {
+        this.target = target;
+    }
+
+    /**
+     * CHANGE_BRANCH ant style matching.
+     * @return target
+     * @see ChangeRequestSCMHead2#getOriginName()
+     */
+    public String getBranch() {
+        return branch;
+    }
+
+    @DataBoundSetter
+    public void setBranch(String branch) {
+        this.branch = branch;
+    }
+
+    /**
+     * CHANGE_FORK ant style matching.
+     * @return fork
+     * @see SCMHeadOrigin.Fork#getName()
+     */
+    public String getFork() {
+        return fork;
+    }
+
+    @DataBoundSetter
+    public void setFork(String fork) {
+        this.fork = fork;
+    }
+
+    /**
+     * CHANGE_URL regular expression matching.
+     *
+     * Ant style matching doesn't work on url strings due to it being meant for regular paths
+     * and all those extra characters conflicts.
+     *
+     * @return urlX
+     * @see ObjectMetadataAction#getObjectUrl()
+     */
+    public String getUrlX() {
```

Comment:
I don't like the `X` suffix on regex fields - `urlPattern` instead perhaps?

Reply:
`pattern` feels more like ant to me, maybe just because I call it that in `branch` and `tag`.
But I needed something simple to distinguish the various comparator types.

Reply:
Another approach could be to have another parameter to specify the comparator to use for all parameters (defaults to ant) but not all things works with ant style matching; url for example.

Reply:
`urlRegexp` maybe, then? `urlX` is definitely not gonna work - short/simple is good, but that's just confusing.